test/test_statement.rb

```diff
@@ -127,7 +127,7 @@ def test_bind_blob
       row = @db.execute('select * from foo')
 
       assert_equal ['hello'], row.first
-      assert_equal row.first.types, ['BLOB']
+      assert_equal ['blob'], row.first.types
```

Comment:
I think I'm confused now... Some types were capitalized before but not all of them???

Reply:
I did this a while back, so frankly I don't remember the details off the top of my head.  But you can see my comments in #2 in my original description.   I was seeing case inconsistency across different operating systems.

Reply:
@casperisfine You can see an action that demonstrates the issue here - https://github.com/sparklemotion/sqlite3-ruby/runs/6090043254 .  Ubuntu passes (lowercase) but Windows/Mac does not.  This was ultimately all coming from the underlying C code, whose behavior didn't seem to be consistent across operating systems.  So I had the Ruby code case normalize, based on what had been in the specs (assuming that was the historically expected behavior)

Reply:
Ok, thanks for the info, I'll try to dig into this a bit more next week. We also need to look at what the common callers expects (e.g. Active Record)

Reply:
@casperisfine Looking it ActiveRecord's type matching, it looks like it's explicitly case insensitive - https://github.com/rails/rails/blob/74ba52ec5c5aa53e2e728d88c179b2d487fff2b4/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L736